libraries/AP_HAL/utility/RingBuffer.h

```diff
@@ -72,6 +72,15 @@ class ByteBuffer {
       read len bytes without advancing the read pointer
     */
     uint32_t peekbytes(uint8_t *data, uint32_t len);
+
+    // Similar to peekbytes(), but will fill out IoVec struct with
+    // both parts of the ring buffer if wraparound is happpening, or
+    // just one part. Returns the number of parts written to.
+    struct IoVec {
```

Comment:
Why not use 'struct iovec' and avoid the "hard cast" from IoVec to struct iovec on the 2 only usages of this function?